src/test/java/com/google/cloud/spanner/jdbc/JdbcDriverTest.java

```diff
@@ -64,7 +74,27 @@ public class JdbcDriverTest {
   public static void startStaticServer() throws IOException {
     MockSpannerServiceImpl mockSpanner = new MockSpannerServiceImpl();
     InetSocketAddress address = new InetSocketAddress("localhost", 0);
-    server = NettyServerBuilder.forAddress(address).addService(mockSpanner).build().start();
+    server =
+        NettyServerBuilder.forAddress(address)
+            .addService(mockSpanner)
+            .intercept(
+                new ServerInterceptor() {
+                  @Override
+                  public <ReqT, RespT> Listener<ReqT> interceptCall(
+                      ServerCall<ReqT, RespT> call,
+                      Metadata headers,
+                      ServerCallHandler<ReqT, RespT> next) {
+                    String clientLibToken =
+                        headers.get(
+                            Metadata.Key.of("x-goog-api-client", Metadata.ASCII_STRING_MARSHALLER));
```

Comment:
Just curious to understand where is this header value actually set?

Reply:
It is normally automatically set by Gax, but in the Spanner client we set it manually to be able to add custom headers:
1. It's defined here: https://github.com/googleapis/sdk-platform-java/blob/43de5b568dc0bbdaddf2419652e0cb16de77ddb6/java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L706
2. It is set here: https://github.com/googleapis/java-spanner/blob/440c88bd67e1c9d08445fe26b01bf243f7fd1ca4/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java#L301